clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -645,6 +647,11 @@ private Collection<SelectionKey> determineHandlingOrder(Set<SelectionKey> select
     private void attemptRead(SelectionKey key, KafkaChannel channel) throws IOException {
         //if channel is ready and has bytes to read from socket or buffer, and has no
         //previous receive(s) already staged or otherwise in progress then read from it
+        log.trace("Attempt read from the channel (channel.ready:{}, key.isReadable:{}, " +
+            "channel.hasBytesBuffered:{}, hasStagedReceive:{}, explicitlyMutedChannels.contains(channel): {})",
+            channel.ready(), key.isReadable(), channel.hasBytesBuffered(), hasStagedReceive(channel),
```

Comment:
if we are adding logs here in selector anyway, do you think it makes sense to add log when it puts receives to the completedReceive queue here https://github.com/linkedin/kafka/pull/370/files#diff-dcc421138f256b5d62a0940bebc779c60d94b5d1d8993456fee52b8d3296e146L1003 ? Basically the layer above it (NetworkClient) will only be dealing with CompletedReceives and though unlikely, there might be cases like muted channel such that the responses are not added to the completedreceives?

Reply:
Yes, I added another log when a NetworkReceive is put into the completed receives.